packages/backend-core/src/sql/sql.ts

```diff
@@ -1236,10 +1270,18 @@ class InternalBuilder {
       }
     }
 
-    // if counting, use distinct count, else select
-    query = !counting
-      ? query.select(this.generateSelectStatement())
-      : this.addDistinctCount(query)
+    const aggregations = this.query.resource?.aggregations || []
+    if (counting) {
+      query = this.addDistinctCount(query)
+    } else if (aggregations.length > 0) {
+      query = query.select(
+        this.knex.raw("ROW_NUMBER() OVER (ORDER BY (SELECT 0)) as _id")
+      )
```

Comment:
This is to give the aggregation rows a unique ID. It's a hack for the rest of the codebase because we make some assumptions that rows have a unique ID. I think the frontend uses it to know what rows to update when updates come through the websockets, though that will never happen for these rows.

This plus the large number of other changes in this PR make me think it's all a bit _too_ hacky and we should make more explicit affordances for "aggregation rows" (rows that do not have an ID).

Reply:
I think while this is hacky its a pretty good idea to get the DB to allocate these numbers, I'm not against continuing down this path.

There is a beneficial piece of work we could do that would make this easier however, right now all tables are assumed to have full CRUD capabilities. If we had the ability to make tables as "read-only" then they wouldn't need to have primary keys - a more significant project but this would also unlock the ability to import views to Budibase which are read-only and don't have primary keys.